lexibank_chacontukanoan.py

```diff
@@ -0,0 +1,145 @@
+from pathlib import Path
+import pylexibank
+
+from clldutils.misc import slug
+
+from pylexibank.util import getEvoBibAsBibtex
+from segments import Tokenizer
+
+
+# Customize your basic data.
+# if you need to store other data in columns than the lexibank defaults, then over-ride
+# the table type (pylexibank.[Language|Lexeme|Concept|Cognate|]) and add the required columns e.g.
+#
+# import attr
+#
+# @attr.s
+# class Concept(pylexibank.Concept):
+#    MyAttribute1 = attr.ib(default=None)
+
+
+class Dataset(pylexibank.Dataset):
+    dir = Path(__file__).parent
+    id = "chacontukanoan"
+
+    # register custom data types here (or language_class, lexeme_class, cognate_class):
+    # concept_class = Concept
+
+    # define the way in which forms should be handled
+    # form_spec = pylexibank.FormSpec(
+    #     brackets={"(": ")"},  # characters that function as brackets
+    #     separators=";/,",  # characters that split forms e.g. "a, b".
+    #     missing_data=('?', '-'),  # characters that denote missing data.
+    #     strip_inside_brackets=True,
+    #     first_form_only = True
+    #     # do you want data removed in brackets or not?
+    # )
+
+    def cmd_download(self, args):
+        """
+        Download files to the raw/ directory. You can use helpers methods of `self.raw_dir`, e.g.
+        to download a temporary TSV file and convert to persistent CSV:
+
+        >>> with self.raw_dir.temp_download("http://www.example.com/e.tsv", "example.tsv") as data:
+        ...     self.raw_dir.write_csv('template.csv', self.raw_dir.read_csv(data, delimiter='\t'))
+        """
+        with self.raw_dir.temp_download(
+                "http://edictor.digling.org/triples/get_data.py?file=tukano",
+                "tukano.tsv") as data:
+            self.raw_dir.write_csv('tukano.csv',
+                                   self.raw_dir.read_csv(data, delimiter='\t'))
+        self.raw_dir.write('sources.bib', getEvoBibAsBibtex('Chacon2014'))
+
+    def cmd_makecldf(self, args):
+        """
+        Convert the raw data to a CLDF dataset.
+
+        A `pylexibank.cldf.LexibankWriter` instance is available as `args.writer`. Use the methods
+        of this object to add data.
+        """
+        data = self.raw_dir.read_csv('tukano.csv', dicts=True)
+        ortho_profile = self.orthography_profile_dict[None]
+        tokenizer = Tokenizer(profile=ortho_profile)
```

Comment:
Yes, here it is: since you use the tokenizer explicitly, and not as part of the class, as it is loaded anyway, you lose normalization. Our tokenizer is a bit more explicit. @chrzyki or @tresoldi do you remember how I get the "normal" tokenizer from within the args/class of a dataset? If you don't know, never mind, and I check in the pylexibank source code tomorrow, but basically, @XachaB, you don't have to do this explicitly, as the tokenizer is an attribute of the dataset class.

Reply:
This is actually the solution we came up with, because using directly the class tokenizer requires to pass an empty dictionary to the first argument (the `item`), as it expects the complete CLDF table row. It could be done, of course, but the code looked a bit more cryptic in our view.

The `None` indexing for the `orthography_profile_dict` (necessary since the multiple profile support) might be a bit obscure if one is not used to `pylexibank`, but I suppose it is a better solution and one could always include a short comment explaining the rationale...

Reply:
@LinguList: Ok, this clarifies my problem.

There is a `self.tokenizer` in the instance, which is defined as a lazy property. But contrary to what the name suggests, it is not directly a `Tokenizer` instance, but instead a function wrapping it and expecting an `item` first argument (a dict). As @tresoldi  said, passing it empty sounded like a hack and we thought probably not the intended usage. 

Depending on what you think is more coherent with the rest of the code base, I can either use `self.tokenizer`, or I can simply normalize by hand before passing the segments to the tokenizer I instantiate now. 

The `None` index in the dictionnary profile dict really threw me off, I thought I had found a bug. You are right,  @tresoldi, I will add a comment if I keep this way of instanciating the profile.